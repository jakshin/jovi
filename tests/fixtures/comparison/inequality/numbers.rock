(Using the inequality comparison operator on numbers)

Int is 42
Float is 12.34
NegativeInt is -42
NegativeFloat is -12.34
ZeroInt is 0
ZeroFloat is 0.0

Say "numbers can equal booleans"
Rightness is right
Wrongness is wrong
say Int is not Rightness
say Int is not Wrongness
say Float is not Rightness
say Float is not Wrongness
say NegativeInt is not Rightness
say NegativeInt is not Wrongness
say NegativeFloat is not Rightness
say NegativeFloat is not Wrongness
say ZeroInt is not Rightness
say ZeroInt is not Wrongness
say ZeroFloat is not Rightness
say ZeroFloat is not Wrongness
say true is not 42
say true is not -42
say true is not -0
say true is not -0.0
say false is not 42
say false is not -42
say false is not -0
say false is not -0.0

Say "integers equal themselves"
say Int ain't Int
say NegativeInt ain't NegativeInt
say ZeroInt ain't ZeroInt
say Int ain't ZeroInt
say NegativeInt ain't ZeroInt
say 42 ain't 42
say -42 ain't -42
say -0 ain't ZeroInt

Say "floats equal themselves"
say Float aint Float
say NegativeFloat aint NegativeFloat
say ZeroFloat aint ZeroFloat
say Float aint ZeroFloat
say NegativeFloat aint ZeroFloat
say 12.34 aint 12.34
say -12.34 aint -12.34
say -0.0 aint ZeroFloat

Say "integers can equal floats"
say Int is not 042.00
say NegativeInt is not -042.00
say Int is not 43.0
say 0 is not ZeroFloat
say 0.0 is not ZeroInt
say -0 is not ZeroFloat
say -0.0 is not ZeroInt
say 43.0 is not 42
