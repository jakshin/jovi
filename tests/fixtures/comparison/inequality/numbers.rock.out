numbers can equal booleans
false
true
false
true
false
true
false
true
true
false
true
false
false
false
true
true
true
true
false
false
integers equal themselves
false
false
false
true
true
false
false
false
floats equal themselves
false
false
false
true
true
false
false
false
integers can equal floats
false
false
true
false
false
false
false
true
